packages/utils/src/lib/report.ts

```diff
@@ -161,6 +167,103 @@ export function countCategoryAudits(
   }, 0);
 }
 
+export function getAuditsFromAllPlugins(
+  report: ScoredReport,
+): EnrichedAuditReport[] {
+  return report.plugins.reduce<EnrichedAuditReport[]>(
+    (acc, plugin) => [...acc, ...plugin.audits],
+    [],
+  );
+}
+
+export function getAuditByRef(
+  { slug, weight, plugin }: CategoryRef,
+  plugins: ScoredReport['plugins'],
+): WeighedAuditReport {
+  const auditPlugin = plugins.find(
+    ({ slug }) => slug === plugin,
+  ) as PluginReport;
+  const audit = auditPlugin?.audits.find(
+    ({ slug: auditSlug }) => auditSlug === slug,
+  );
+  if (!audit) {
+    throwIsNotPresentError(`Audit ${slug}`, auditPlugin?.slug);
+  }
+  return {
+    ...audit,
+    weight,
+    plugin,
+  };
+}
+
+export function getGroupWithAudits(
+  refSlug: string,
+  refPlugin: string,
+  plugins: ScoredReport['plugins'],
+): EnrichedScoredAuditGroupWithAudits {
+  const plugin = plugins.find(({ slug }) => slug === refPlugin) as PluginReport;
+  const groupWithAudits = plugin?.groups?.find(
+    ({ slug }) => slug === refSlug,
+  ) as EnrichedScoredAuditGroupWithAudits;
+
+  if (!groupWithAudits) {
+    throwIsNotPresentError(`Group ${refSlug}`, plugin?.slug);
+  }
+  const groupAudits = groupWithAudits.refs.reduce<WeighedAuditReport[]>(
+    (acc: WeighedAuditReport[], ref) => {
+      const audit = getAuditByRef(
+        { ...ref, plugin: refPlugin } as CategoryRef,
+        plugins,
+      );
+      if (audit) {
+        return [...acc, audit];
+      }
+      return [...acc];
+    },
+    [],
+  ) as WeighedAuditReport[];
+  const audits = groupAudits.sort(sortCategoryAudits);
+
+  return {
+    ...groupWithAudits,
+    audits,
+  };
+}
+
+export function sortCategoryAudits(
```

Comment:
It would be valuable to document the sorting logic by creating unit tests for it.

Reply:
Added tests